Source/WebCore/html/HTMLFormControlElement.h

```diff
@@ -38,13 +38,13 @@ class HTMLFormControlElement : public LabelableElement, public ValidatedFormList
 public:
     virtual ~HTMLFormControlElement();
 
-    bool isValidatedFormListedElement() const override { return true; }
-    bool isFormListedElement() const override { return true; }
+    bool isValidatedFormListedElement() const final { return true; }
+    bool isFormListedElement() const final { return true; }
 
-    bool matchesValidPseudoClass() const override { return willValidate() && isValidFormControlElement(); }
-    bool matchesInvalidPseudoClass() const override { return willValidate() && !isValidFormControlElement(); }
+    bool matchesValidPseudoClass() const override { return ValidatedFormListedElement::matchesValidPseudoClass(); }
```

Comment:
Can we just remove these overrides?

Reply:
Unfortunately, no: this `matchesValidPseudoClass()` is a virtual method, defined on `Element`'s vtable, while `ValidatedFormListedElement`'s counterpart is non-virtual and the class doesn't inherit `Element`.

Also, we can't do `using ValidatedFormListedElement::matchesValidPseudoClass` (I've tried, tests are failing).

Thank you for review!  🙇